pephub/templates/base/edit_project.html

```diff
@@ -135,6 +140,65 @@
   const form = document.getElementById("metadata-form")
   form.addEventListener("change", detectMetadataChanges)
 
+  // submit PATCH request to server/database
+  const handleMetaMetaDataSubmit = () => {
+
+    // update save btn for UX and feedback
+    const saveBtn = document.getElementById("metadata-save-btn")
+    saveBtn.innerHTML = "Saving..."
+    saveBtn.disabled = true
+
+    // send request to server
+    fetch("/api/v1/projects/{{namespace}}/{{ project.name }}?tag={{ tag }}", {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: document.getElementById("project-name").value,
+        tag: document.getElementById("project-tag").value,
+        description: document.getElementById("project-description").value,
+        private: document.getElementById("is-private-toggle").checked,
+      }),
+    })
+      .then((response) => {
+        if (response.ok) {
+          // update original values
+          setOriginalProjectNameValue(document.getElementById("project-name").value)
+          setOriginalProjectNameValue(document.getElementById("project-tag").value)
```

Comment:
You're using the same setter here twice. Probably need a new `setOriginalProjectTag` function. 

FYI, we might want @nsheff 's eyes on some of this. I really am out of my element messing around with state management in vanilla JS.

Reply:
yeah, I see, for tag it will be same function, we are changing similar thing. For me it was working

Reply:
Hmm. Those setters are for storing local state to change the HTML/CSS of the form for user feedback to let them know if they can submit. I.e., if they made change, the "Save" button becomes enabled. If that makes sense.

Reply:
Its ok. I need to clean up this javascript anyways, so we can merge and I will tweak it.

Reply:
I didn't know about setters..., yeah, you can just merge and work on it